### PR TITLE
feat: add aws_security_group_rule_deprecated rule

### PR DESCRIPTION
**Pull request by kayman-mk**
Adds a rule which generates a warning if a `aws_security_group_rule` is used. According to the docs the `aws_vpc_security_group_egress_rule` or `aws_vpc_security_group_ingress_rule` should be used.

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule

### CODE REVIEW

File: docs/rules/aws_security_group_rule_deprecated.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# aws_security_group_rule_deprecated
+
+// TODO: Write the rule's description here
```
Comment (bendrucker) [inline docs/rules/aws_security_group_rule_deprecated.md:3]: TODO

File: docs/rules/aws_security_group_rule_deprecated.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Why
+
+Avoid using the `aws_security_group_rule` resource, as it struggles with managing multiple CIDR blocks, and, due to the historical lack of unique IDs, tags and descriptions.
+
+For further information, see the [Terraform documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule).
+
+## How To Fix
+
+Depending on `foo.type`, you can fix the issue by using either `aws_vpc_security_group_egress_rule` or `aws_vpc_security_group_ingress_rule`.
```
Comment (bendrucker) [inline docs/rules/aws_security_group_rule_deprecated.md:31]:
```suggestion
Depending on `type`, you can fix the issue by using either `aws_vpc_security_group_egress_rule` or `aws_vpc_security_group_ingress_rule`.
```

File: rules/aws_security_group_rule_deprecated.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+// Link returns the rule reference link
+func (r *AwsSecurityGroupRuleDeprecatedRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check that aws_security_group_rule resource is not used
+func (r *AwsSecurityGroupRuleDeprecatedRule) Check(runner tflint.Runner) error {
+	resources, err := runner.GetResourceContent(r.resourceType, &hclext.BodySchema{
+		Attributes: []hclext.AttributeSchema{
+			{Name: r.attributeName},
```
Comment (bendrucker) [inline rules/aws_security_group_rule_deprecated.go:49]: Is there even a need to parse attributes here? Seems not, just the existence of the resource should trigger.

Comment (kayman-mk) [reply rules/aws_security_group_rule_deprecated.go:49]: Hm, I don't see this on my machine. It's not used.

**Comment by kayman-mk** [comment]
Everything addressed, docs reworked.

**Comment by kayman-mk** [comment]
Not sure how to decide about "enabled by default" and "severity".

**Review by wata727 — APPROVED** [review]

**Pull request merged**